lib/features/parkings_view/widgets/parking_wide_tile_card.dart

```diff
@@ -143,3 +131,69 @@ class _RightColumn extends StatelessWidget {
     );
   }
 }
+
+class _FavouriteWidget extends StatefulWidget {
+  const _FavouriteWidget({required this.parking});
+  final Parking parking;
+
+  @override
+  State<_FavouriteWidget> createState() => _FavouriteWidgetState();
+}
+
+class _FavouriteWidgetState extends State<_FavouriteWidget> {
+  bool _isFavourite = false;
+
+  Future<bool> _loadPreference() async {
+    final prefs = await SharedPreferences.getInstance();
+    return prefs.getBool(widget.parking.id) ?? false;
+  }
+
+  Future<void> _updatePreference() async {
+    setState(() {
+      _isFavourite = !_isFavourite;
+    });
+    final prefs = await SharedPreferences.getInstance();
+    await prefs.setBool(widget.parking.id, _isFavourite);
+  }
```

Comment:
So basically this whole statefull widget is actually well crafted I think. It's simple logic so you could say it's enough for this case. But I just wanted to show you how you can more separate logic from UI here using riverpod. Maybe in this simple case this isn't really necessary and you solution was pretty much enough, but I think it's much cleaner if you separate disk operation to "local repository" like this:

```dart
part "local_fav_parking_repository.g.dart";

@riverpod
class LocalFavParkingsRepository extends _$LocalFavParkingsRepository {
  static const _repoPrefix =
      "fav_parking_"; // actually if you use shared prefs, it's good idea to use some prefix to avoid conflicts with other features. If you've used a real DB then you would have separate collection/datatable for this feature, but shared prefs is just a bug of key-value pairs so some prefix is a good idea.

  String get _key => _repoPrefix + parkingId;

  @override
  bool? build(String parkingId) {
    final prefs = ref.watch(sharedPreferencesSingletonProvider);
    return switch (prefs) {
      AsyncError() => null,
      AsyncValue(:final value) => value?.getBool(_key) ?? false,
    };
  }

  Future<void> toggle() async { // if you really would want an "enteprise edition" of architecture here, this should only accept bool and save it and toggling logic should be handled in some controller instance (another riverpod layer). But yeah, this would be a real overkill for just one bool xD
    final currState = state ?? false;
    final newState = !currState;
    state = newState;
    final prefs = await ref.read(sharedPreferencesSingletonProvider.future);
    await prefs.setBool(_key, newState);
  }
}
```

then pretty much your widget gets very simplified and only defines how your UI reacts to specific state and doesn't care about handling state. Clear and nice separation of concerns imo. But yeah, I know this feature is not too heavy, but I think it's good idea to always separate those layers.

```dart

class FavouriteParkingWidget extends ConsumerWidget {
  const FavouriteParkingWidget(this.parking);
  final Parking parking;
  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final favController =
        ref.watch(localFavParkingsRepositoryProvider(parking.id).notifier);
    final isFavorite =
        ref.watch(localFavParkingsRepositoryProvider(parking.id));
    return GestureDetector(
      onTap: favController.toggle,
      child: isFavorite == null
          ? const FavouriteIcon(
              icon: Icons.error,
            )
          : FavouriteIcon(
              icon:
                  isFavorite ? Icons.favorite : Icons.favorite_border_outlined,
            ),
    );
  }
}

```

Reply:
tell me what you think of it

Reply:
I like this changes